openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -202,7 +206,8 @@ def get_cohort(user, course_key, assign=True, use_cached=False):
             assignment_type=CourseCohort.RANDOM
         ).course_user_group
 
-    user.course_groups.add(cohort)
+    membership = CohortMembership(course_user_group=cohort, user=user)
```

Comment:
Ideally, I'd like to have `CourseUserGroup` and `User` relate to each other using a `ManyToManyField` with a `through` relationship to use `CohortMembership` as an intermediary. The problem there is that `CohortMembership` only applies to a subset of `CourseUserGroup` objects (those with `group_type=CourseUserGroup.COHORT`), and I don't think we can "filter" a `through` relationship. If we were to link them in this way, we'd end up assuming all `CourseUserGroup` objects were cohorts, which could break things if we ever end up making a different type that may not have the same uniqueness restrictions as `CourseUserGroup.COHORT`.